### PR TITLE
[Method References 2] LPS-166968

### PR DESCRIPTION
**Pull request by JanisZhang**
This PR is to use serviceTrackerMap instead of the method reference in `data-engine-taglib`. Thanks!

### CODE REVIEW

File: modules/apps/data-engine/data-engine-taglib/src/main/java/com/liferay/data/engine/taglib/internal/servlet/taglib/util/DataLayoutTaglibUtil.java [hunk 1]
```diff
@@ -258,90 +258,55 @@ public static String resolveModule(String moduleName) {
...
+			ServiceTrackerMapFactory.openSingleValueMap(
+				bundleContext, DataDefinitionContentType.class, null,
+				(serviceReference, emitter) -> {
+					String contentType = GetterUtil.getString(
+						serviceReference.getProperty("content.type"));
 
-		if (Validator.isNull(contentType)) {
-			return;
-		}
+					if (Validator.isNull(contentType)) {
```
Comment (dantewang) [inline modules/apps/data-engine/data-engine-taglib/src/main/java/com/liferay/data/engine/taglib/internal/servlet/taglib/util/DataLayoutTaglibUtil.java:285]: @JanisZhang Does this logic (if can't get this property, do nothing) match the `openSingleValueMap` method which takes `key` as a parameter?

**Comment by dantewang** [comment]
@JanisZhang Please double check and see if you can simplify this.

**Comment by JanisZhang** [comment]
Hi @dantewang Force-pushed, Thanks! see https://github.com/liferay-core-infra/liferay-portal/pull/2243#discussion_r1009213766

**Comment by dantewang** [comment]
ci:test:sf

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
@JanisZhang no regressions found.

**Comment by JanisZhang** [comment]
@catarineaguiar thank you!

File: modules/apps/data-engine/data-engine-taglib/src/main/java/com/liferay/data/engine/taglib/internal/servlet/taglib/util/DataLayoutTaglibUtil.java [hunk 2]
```diff
@@ -258,90 +257,34 @@ public static String resolveModule(String moduleName) {
...
-		String contentType = GetterUtil.getString(
-			properties.get("content.type"));
-
-		if (Validator.isNull(contentType)) {
-			return;
-		}
-
-		_dataLayoutBuilderDefinitions.put(
-			contentType, dataLayoutBuilderDefinition);
+		_dataDefinitionContentTypeServiceTrackerMap =
```
Comment (tinatian) [inline modules/apps/data-engine/data-engine-taglib/src/main/java/com/liferay/data/engine/taglib/internal/servlet/taglib/util/DataLayoutTaglibUtil.java:262]: @JanisZhang @dantewang Please do not use property "content.type", there is a method getContentType exposed from API providing same information

**Comment by tinatian** [comment]
@JanisZhang As we talked in the meeting, please resend.

Thanks.
Tina

**Pull request closed without merging**